xml/System/Span`1+Enumerator.xml

```diff
@@ -29,17 +29,17 @@ The C# [foreach](~/docs/csharp/language-reference/keywords/foreach-in.md) of the
 
 Enumerators can be used to read the data in the <xref:System.Span%601>, but they cannot be used to modify it. 
 
-Initially, the enumerator is positioned before the first element in the <xref:System.Span%601>. At this position, attempting to retrieve an element from the <xref:System.Span%601.Enumerator.Current> property throws an <xref:System.IndexOutOfRangeException>. To prevent an exception, you must call <xref:System.Span%601.Enumerator.MoveNext%2A> to advance the enumerator to the first item in the <xref:System.Span%601> before reading the value of <xref:System.Span%601.Enumerator.Current>.
+Initially, the enumerator is positioned before the first element in the <xref:System.Span%601>. At this position, <xref:System.Span%601.Enumerator.Current> is undefined. You must call <xref:System.Span%601.Enumerator.MoveNext%2A> to advance the enumerator to the first item in the <xref:System.Span%601> before reading the value of <xref:System.Span%601.Enumerator.Current>.
```

Comment:
Just to confirm, did we decide to change this to "undefined" even though we throw an exception here?

Reply:
My reasoning is that *is undefined* here and in other places communicates to the user of the API: *it's not relevant, you better not to use this API in this state, implementation might change because of some internal reasons, various implementations of that member for different classes differ*. So, yes, to avoid leaking implementation details. With that said, the `Span<T>.Enumerator.Current` page still contains the **Exceptions** section. In that case, the rule to document exceptions overrules intention to hide implementation details.  Any solution looks to be not perfect.

/cc @BillWagner @stephentoub

Reply:
> In that case, the rule to document exceptions overrules intention to hide implementation details.

FWIW, I disagree.  The exceptions document exceptions that can occur for defined behavior.  Undefined behavior is undefined behavior and isn't relevant to any portion of the documentation, including Exceptions.

Reply:
@stephentoub thank you for the reply. That is convincing. @BillWagner thanks for the thumbs-up.  I've removed the exceptions section from the docs of the `Current` property.

Now this PR looks to be ready to merge.